src/validator/IndexScanValidator.cpp

```diff
@@ -153,5 +122,89 @@ Status IndexScanValidator::checkFilter(Expression* expr, const std::string& from
     return Status::OK();
 }
 
+Status IndexScanValidator::checkRelExpr(RelationalExpression* expr,
+                                        const std::string& from) {
+    auto* left = expr->left();
+    auto* right = expr->right();
+    // Does not support filter : schema.col1 > schema.col2
+    if (left->kind() == Expression::Kind::kLabelAttribute &&
+        right->kind() == Expression::Kind::kLabelAttribute) {
+        return Status::NotSupported("Expression %s not supported yet",
+                                    expr->toString().c_str());
+    } else if (left->kind() == Expression::Kind::kLabelAttribute ||
+               right->kind() == Expression::Kind::kLabelAttribute) {
+        auto ret = rewriteRelExpr(expr, from);
+        NG_RETURN_IF_ERROR(ret);
+    } else {
+        return Status::NotSupported("Expression %s not supported yet",
+                                    expr->toString().c_str());
+    }
+    return Status::OK();
+}
+
+Status IndexScanValidator::rewriteRelExpr(RelationalExpression* expr,
+                                          const std::string& from) {
+    auto* left = expr->left();
+    auto* right = expr->right();
+    auto leftIsAE = left->kind() == Expression::Kind::kLabelAttribute;
+
+    std::string ref, prop;
+    auto* la = leftIsAE
+               ? static_cast<LabelAttributeExpression *>(left)
+               : static_cast<LabelAttributeExpression *>(right);
+    if (*la->left()->name() != from) {
+        return Status::SemanticError("Schema name error : %s",
+                                     la->left()->name()->c_str());
+    }
+
+    ref = *la->left()->name();
+    prop = *la->right()->name();
+
+    // rewrite ConstantExpression
+    auto c = leftIsAE
+             ? checkConstExpr(right, prop)
+             : checkConstExpr(left, prop);
+
+    if (!c.ok()) {
+        return Status::SemanticError("expression error : %s", left->toString().c_str());
```

Comment:
`This error message should be Status::SemanticError("expression error : %s", expr->toString().c_str());`
I have corrected it in the next PR.